### ISSUE TITLE
Previously compiled schema reused

### ISSUE THREAD

**Issue by abyssary** [horejsek/python-fastjsonschema#94]
If `compile` is called with two different json schema and the second schemas `$id` field is set to an empty string, the second schema is ignored and the first one is used for all validation calls.

Sample code to demonstrate the behaviour
```
from fastjsonschema import compile

schema1 = {
    "$schema": "http://json-schema.org/draft-07/schema",
    "$id": "",
    "properties": {
        "foo": {
            "type": "string",
        }
    },
    "required": [
        "foo"
    ]
}

schema2 = {
    "$schema": "http://json-schema.org/draft-07/schema",
}

input1 = {"foo": "bar"}
input2 = {}

validate = compile(schema1)
validate(data=input1)

validate = compile(schema2)
validate(data=input2)
```

The last line will throw an exception
```
fastjsonschema.exceptions.JsonSchemaException: data must contain ['foo'] properties
```

fastjsonschema version 2.14.4
python 3.7.5

Setting `$id` on schema1 to any other string or adding `$id` to schema2 (including the empty string) will not trigger the exception.

https://json-schema.org/understanding-json-schema/structuring.html#the-id-property
states that an empty string is not a valid value for '$id' so i'm not sure if this should be considered a bug but the behaviour ist still surprising.

Older versions (at least 2.7) did not show this behaviour.

**Comment by rmorshea** [comment]
Seems like this might be fixed now? https://github.com/horejsek/python-fastjsonschema/blob/master/CHANGELOG.txt#L3

**Comment by abyssary** [comment]
Unfortunately not.
The problem only occurs with the newest version 2.14.4 (2.14.3 works fine), so it seems the change you highlighted is acutally responsible.

**Comment by horejsek** [comment]
Fixed in version 2.14.5.
